File: fixtures/scale/ruby/large/lib/check_carousel.rb
require_relative 'carousel_encrypt'
require_relative 'carousel_decrypt'
require_relative 'corpus_rotate'

def check_carousel
  corpus_rotate.each do |text|
    sealed = carousel_encrypt(text, 18)
    opened = carousel_decrypt(sealed, 18)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_carousel.py
from cipherlab.ciphers.carousel import carousel_encrypt, carousel_decrypt
from cipherlab.specs.carousel_spec import carousel_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_carousel():
    spec = carousel_spec()
    for text in corpus_rotate():
        sealed = carousel_encrypt(text, spec["key"])
        opened = carousel_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"
